apps/dashboard/src/components/engine/configuration/kms-aws-config.tsx

```diff
@@ -1,126 +1,173 @@
+              {...form.register("awsRegion")}
             />
-          </FormControl>
-          <FormControl isRequired>
-            <FormLabel>Secret Key</FormLabel>
+          </FormFieldSetup>
+
+          <FormFieldSetup
+            label="Access Key"
+            errorMessage={
+              form.getFieldState("awsAccessKeyId", form.formState).error
+                ?.message
+            }
+            htmlFor="aws-access-key"
+            isRequired={false}
+            tooltip={null}
+          >
             <Input
-              placeholder="e.g. UW7A..."
+              id="aws-access-key"
+              placeholder="AKIA..."
               autoComplete="off"
               type="text"
-              {...form.register("awsSecretAccessKey", { required: true })}
+              {...form.register("awsAccessKeyId")}
             />
-          </FormControl>
-        </Flex>
-        <FormControl isRequired>
-          <FormLabel>Region</FormLabel>
-          <Input
-            w={{ base: "full", md: "49%" }}
-            placeholder="e.g. us-west-1"
-            autoComplete="off"
-            type="text"
-            {...form.register("awsRegion", { required: true })}
-          />
-        </FormControl>
-      </Card>
-      <Flex justifyContent="end" gap={4} alignItems="center">
-        {form.formState.isDirty && (
-          <Text color="red.500">
-            This will reset your other backend wallet configurations
-          </Text>
-        )}
-        <Button
-          isDisabled={!form.formState.isDirty}
-          w={{ base: "full", md: "inherit" }}
-          colorScheme="primary"
-          px={12}
-          type="submit"
-        >
-          {form.formState.isSubmitting ? "Saving..." : "Save"}
-        </Button>
-      </Flex>
-    </Flex>
+          </FormFieldSetup>
+
+          <FormFieldSetup
+            label="Secret Key"
+            errorMessage={
+              form.getFieldState("awsSecretAccessKey", form.formState).error
+                ?.message
+            }
+            htmlFor="aws-secret-key"
+            isRequired={false}
+            tooltip={null}
+          >
+            <Input
+              id="aws-secret-key"
+              placeholder="UW7A..."
+              autoComplete="off"
+              type="text"
+              {...form.register("awsSecretAccessKey")}
+            />
+            <FormDescription className="pt-2">
+              This will not be shown again.
+            </FormDescription>
+          </FormFieldSetup>
+        </div>
+
+        <div className="flex items-center justify-end gap-4">
+          {!supportsMultipleWalletTypes && (
+            <p className="text-destructive-text text-sm">
+              This will clear other credentials.
+            </p>
+          )}
+          <Button disabled={!form.formState.isDirty} type="submit">
+            {form.formState.isSubmitting ? "Saving..." : "Save"}
```

Comment:
Render a `<Spinner className='size-4' />` on left side of Saving. ( You'll also have to add `gap-2` className on `Button`

Reply:
Makes sense, this was old code from the previous form behavior.